src/bdecode.cpp

```diff
@@ -479,7 +479,7 @@ namespace libtorrent
 		{
 			bdecode_token const& t = tokens[token];
 			TORRENT_ASSERT(t.type == bdecode_token::string);
-			int const size = m_root_tokens[token + 1].offset - t.offset - t.start_offset();
```

Comment:
for msvc release replacing "m_root_tokens" with "tokens" generates good code (ms deduce how to optimal calculate address), but mingw gcc still remains unoptimized. i will post later gcc mingw disasm